web-ui/src/main/java/org/archcnl/ui/input/ruleeditor/ArchitectureRulesContract.java

```diff
@@ -0,0 +1,16 @@
+package org.archcnl.ui.input.ruleeditor;
+
+import java.io.Serializable;
+
+public interface ArchitectureRulesContract {
+
+    public interface View {
+        public void saveRule();
```

Comment:
The contract does not act as an interface description for the View and the Presenter. Its only purpose is to describe which actions the view is allowed to perform on the presenter and the other way around.
The method `saveRule` is only meant to be called from within the view and never from the presenter. This means that it should not be part of the contract between the View and the Presenter.

Reply:
I thought that maybe in the future there is a scenario where the presenter might want to read out the textarea. But thinking about it it seems unlikely. In any case the method should only be part of the contract in the case that the presenter uses it, so I removed it.